### ISSUE TITLE
Upgrade to Raven 3.0

### ISSUE THREAD

**Issue by kblooie** [NEventStore/NEventStore.Persistence.RavenDB#10]

**Comment by Iridio** [comment]
I have already done the migration, but it's necessary the framework 4.5. So befaore doing a PR I was waiting to know when NES will move to 4.5

**Comment by Iridio** [comment]
I have already done the migration, but it's necessary the framework 4.5. So before doing a PR I was waiting to know when NES will move to 4.5

**Comment by kblooie** [comment]
Ha, I did as well. 

All of NES doesn't have to move to .NET 4.5, we just have to ensure that it has a 4.5 target along with 4.0 in the nuget package. I thought I added that a while back. If not, I'll add it today.

More importantly, this will be a breaking change for people using .NET 4.0 or running on Raven 2.5. The checkpoint api change you contributed (thanks a bunch by the way) could be important to people who are on 4.0 or raven 2.5. Early on, we considered releasing a v5 and v6 version of the raven package at the same time to tackle this. Which is why all the persistence engines got split out into their own repos.

Who thinks this is needed?
